lib/anony/anonymisable.rb

```diff
@@ -63,16 +63,20 @@ def valid_anonymisation?
     # @example
     #   manager = Manager.first
     #   manager.anonymise!
+    # rubocop:disable Metrics/AbcSize
     def anonymise!
       unless self.class.anonymise_config
         raise ArgumentError, "#{self.class.name} does not have an Anony configuration"
       end
 
       self.class.anonymise_config.validate!
       self.class.anonymise_config.apply(self)
+    rescue ActiveRecord::RecordInvalid => e
+      raise Anony::RecordInvalid, "#{self.class.name} - raised #{e.class.name} #{e.message}"
```

Comment:
I'm a little conflicted about whether we should do this here; the caller should know the model that they were working with when they called `anonymise!` so can add any extra context around that, and as things stand they'd have access to the full ActiveRecord exception so can pull anything they want from it.

Should we be instead returning `Result.failed(e)` here? And making sure the caller logs out the class/ID if needed?

Reply:
I agree. Can we do anything useful with `Exception#cause` instead which allows us to access the original error? (which includes the model itself I think...)

Reply:
> Should we be instead returning Result.failed(e) here? And making sure the caller logs out the class/ID if needed?